006-strimzi-ui.md

```diff
@@ -0,0 +1,178 @@
+Given I am a user of the UI
+When I navigate to the topic listing
+And I filter for topic 'SampleTopic'
+Then I see topic 'SampleTopic' in the topic list
+And Topic 'SampleTopic' has '3' partitions and '2' replicas
+
+Scenario: Viewing a specific topic (when it does not exist in the cluster):
+Given I am a user of the UI
+When I navigate to the topic listing
+And I filter for topic 'SampleTopic'
+Then I am told topic 'SampleTopic' does not exist in the cluster
+```
+
+#### Follow on capabilities
+
+Additional Kafka capabilities which could be added later include (but would not be limited to):
+
+- Create, update and delete topics on the topics list page (recontextualised as a Topics page - as per mock up)
+- View Consumer group status for a given topic, or the whole cluster
+- View the Brokers in a given cluster, their configuration, and where appropriate, allow modification of broker configuration
+- Provide details of Kafka listener/bootstrap addresses, along with sample configuration, to allow streamline client connectivity
+
+If desired, additional Strimzi provided capabilities, such as Kafka Users, could be leveraged/added/managed through this UI also (for example, allowing access to the UI and its capabilities, general CRUD updates to them etc). 
+
+Finally, one capability which may be of interest (which may have bearing on how this is surfaced in Strimzi itself) is having a single UI, which can manage multiple Kafka clusters deployed via the Strimzi operator.
+
+In all of these cases, capabilities can be added in a prioritised order, and should be added in a phased manner themselves (for example, add a view of the brokers and their configuration, then the ability to modify select configuration).
+
+#### Proposed deployment
+
+This UI could be deployed as a part of Strimzi as follows:
+
+![Suggested deployment](./images/006-deployment.png)
+
+Where:
+
+- The UI is deployed standalone, akin to how the Kafka bridge, Kafka Connect are currently for example (so a new CRD would be required, with configuration needed to reference which cluster(s) to interact with).
+- The UI's spec would contain 0 to N Kafka clusters the UI will operate against, which will: 
+  - Contain 1 or more 'backend' entries. These represent backend sources of data which can be surfaced in the UI.
+    - One of these entries will be the `strimzi-api` (ie each Kafka kind will contain its own api server).
+  - Each of these entries will define items such as the address to use to connect to it, any auth credentials required, Transport security required, etc.
+- The UI's spec will contain general configuration for the UI - such as certificates to expose to clients etc.
+
+This could look as follows in a CR configured by a user (note that common fields such as image, or readiness and liveness probes have been omitted here for conciseness, but would be present in a full CR). Comments are included inline for clarification of intent/type:
+
+```
+...
```

Comment:
I guess this should be the new Kafka web UI custom resource, can you be more explicit with the lines related to apiVersion, kind and so on please?

Reply:
Sure - I will add these in. I wanted to highlight the items a UI would need to know to operate (with the rest as a given). But more than happy to be more explicit